crates/cargo-test-support/src/lib.rs

```diff
@@ -470,24 +491,8 @@ pub fn main_file(println: &str, deps: &[&str]) -> String {
     buf
 }
 
-// Path to cargo executables
-pub fn cargo_dir() -> PathBuf {
-    env::var_os("CARGO_BIN_PATH")
-        .map(PathBuf::from)
-        .or_else(|| {
-            env::current_exe().ok().map(|mut path| {
-                path.pop();
-                if path.ends_with("deps") {
-                    path.pop();
-                }
-                path
-            })
-        })
-        .unwrap_or_else(|| panic!("CARGO_BIN_PATH wasn't set. Cannot continue running test"))
-}
-
 pub fn cargo_exe() -> PathBuf {
-    cargo_dir().join(format!("cargo{}", env::consts::EXE_SUFFIX))
+    snapbox::cmd::cargo_bin("cargo")
```

Comment:
This doesn't seem to be implemented quite the same. Is there a particular reason that it doesn't use `CARGO_BIN_PATH`? Making assumptions around `current_exe` seems potentially risky, and something we'd like to discourage.

Reply:
I'm not seeing anything set `CARGO_BIN_PATH` unless that is an undocumented escape hatch for developers to override where it comes from.  I just double checked that its not being used on my system and that instead we are falling back to the `current_exe` trick that already exists in `cargo-test-support`:
```
[crates/cargo-test-support/src/lib.rs:475] env::var_os("CARGO_BIN_PATH") = None
[crates/cargo-test-support/src/lib.rs:483] path = "/home/epage/src/personal/cargo/target/debug"
```

Reply:
Oh, I guess CARGO_BIN_PATH is an old thing from long ago. I guess it could be used to test with a different cargo, but I've never used that.  I was thinking it was `CARGO_BIN_EXE_cargo`, which I thought we switched to long ago, but I guess we didn't (and I don't think it works anyways since this code lives in a dependency).